lib/taskana-core/src/main/java/pro/taskana/common/internal/configuration/SecurityVerifier.java

```diff
@@ -0,0 +1,97 @@
+package pro.taskana.common.internal.configuration;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+import org.apache.ibatis.jdbc.SqlRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pro.taskana.common.api.exceptions.SystemException;
+
+public class SecurityVerifier {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SecurityVerifier.class);
+  private static final String SECURITY_FLAG_COLUMN_NAME = "ENFORCE_SECURITY";
+  private static final String INSERT_SECURITY_FLAG = "INSERT INTO %s.CONFIGURATION VALUES (%b)";
+  private static final String SELECT_SECURITY_FLAG = "SELECT %s FROM %s.CONFIGURATION";
+  private final String schemaName;
+  private DataSource dataSource;
+
+
+  public SecurityVerifier(DataSource dataSource, String schema) {
+    super();
+    this.dataSource = dataSource;
+    this.schemaName = schema;
+  }
+
+  public void checkSecureAccess(boolean securityEnabled) {
+
+    if (!securityEnabled) {
```

Comment:
why do you check this? Even if security is activated we might be the first one and want to insert this information into the database, don't we?